packages/hub/writers.js

```diff
@@ -26,11 +28,13 @@ class Writers {
     );
     try {
       let newSchema = await schema.validate(pending, { type, session });
-      let response = await this._finalizeAndReply(pending);
+      let context = await this._finalizeAndReply(pending, branch, session, type, schema);
       if (newSchema) {
         this.schema.invalidateCache();
       }
-      return response;
+
+      // TODO index the search doc
```

Comment:
@ef4 this is the place where I'm not sure what the correct approach should be to update the index. should I use the `hub:indexers.update()`?

Reply:
You should [inject the pgsearch client](https://github.com/cardstack/cardstack/blob/cc237da819a33c9c50eb5f8d059c6adafaa4f8f3/packages/hub/indexers.js#L50) and use its [`saveDocument`](https://github.com/cardstack/cardstack/blob/cc237da819a33c9c50eb5f8d059c6adafaa4f8f3/packages/pgsearch/client.js#L120)  and `deleteDocument` methods.

When you do that I think you will find that it's worth refactoring `saveDocument` so that it accepts a `DocumentContext` instance as its argument, because most of what it takes now is things that are available there, and it already takes too many arguments.

When you save from `hub:writers`, there is no `generation` to worry about, you can leave that empty.

Reply:
sweet thanks!